From 8c33e9812ffeef92e9fd6f70e927358087014690 Mon Sep 17 00:00:00 2001
From: rsjtdrjgfuzkfg <public@rsjtdrjgfuzkfg.com>
Date: Tue, 9 Jan 2018 07:24:41 +0100
Subject: [PATCH] Fix building on GCC 6.4

This commit might break building in older versions of GCC.
---
 build.sh                                             | 2 +-
 components/dbengine/src/DatabaseResult.cpp           | 4 ++--
 components/devices/device/src/sbRequestThreadQueue.h | 1 +
 3 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/build.sh b/build.sh
index a2b0dcd504..0d4649ca53 100755
--- a/build.sh
+++ b/build.sh
@@ -77,7 +77,7 @@ case $OSTYPE in
         depdate="20130316"
         fname="$depdirn-$version-$depdate-$build-final.tar.lzma"
 
-        export CXXFLAGS="-O2 -fomit-frame-pointer -Wno-delete-non-virtual-dtor -Wno-unused-but-set-variable $CXXFLAGS $CPPFLAGS"
+        export CXXFLAGS="-O2 -fomit-frame-pointer -Wno-delete-non-virtual-dtor -Wno-unused-but-set-variable -D_GLIBCXX_USE_CXX11_ABI=0 $CXXFLAGS $CPPFLAGS"
 
         echo "linux $arch"
         ( cd dependencies && {
diff --git a/components/dbengine/src/DatabaseResult.cpp b/components/dbengine/src/DatabaseResult.cpp
index 02bf617920..c78dffc433 100644
--- a/components/dbengine/src/DatabaseResult.cpp
+++ b/components/dbengine/src/DatabaseResult.cpp
@@ -400,7 +400,7 @@ void CDatabaseResult::RebuildColumnResolveMap()
 
       PRUint32 nSize =  m_ColumnNames.size();
       for(PRUint32 i = 0; i < nSize; i++) {
-        m_ColumnResolveMap.insert(std::make_pair<nsString, PRUint32>(m_ColumnNames[i], i));
+        m_ColumnResolveMap.insert(std::make_pair(m_ColumnNames[i], i));
       }
     }
     
@@ -412,7 +412,7 @@ void CDatabaseResult::RebuildColumnResolveMap()
 
     PRUint32 nSize =  m_ColumnNames.size();
     for(PRUint32 i = 0; i < nSize; i++) {
-      m_ColumnResolveMap.insert(std::make_pair<nsString, PRUint32>(m_ColumnNames[i], i));
+      m_ColumnResolveMap.insert(std::make_pair(m_ColumnNames[i], i));
     }
   }
 }
diff --git a/components/devices/device/src/sbRequestThreadQueue.h b/components/devices/device/src/sbRequestThreadQueue.h
index 0191f40eef..8bb70526bd 100644
--- a/components/devices/device/src/sbRequestThreadQueue.h
+++ b/components/devices/device/src/sbRequestThreadQueue.h
@@ -84,6 +84,7 @@ class sbRequestThreadQueue
     typedef RequestItems::size_type size_type;
     typedef RequestItems::reference reference;
     typedef RequestItems::const_reference const_reference;
+    typedef RequestItems::value_type value_type;
 
     Batch();
     ~Batch();
